library/systemd/src/lib/yast2/system_service.rb

```diff
@@ -406,16 +412,12 @@ def changed?(key = nil)
     #
     # @param value [Boolean] true to set this service as active
     def active=(value)
-      if value == currently_active?
-        unregister_change(:active)
-      else
-        register_change(:active, value)
-      end
+      register_change(:active, value)
     end
 
     # Sets start mode to the underlying system
     def save_start_mode
-      return unless changes[:start_mode]
+      return if changes[:start_mode].nil? || changes[:start_mode] == current_start_mode
```

Comment:
The advantage of checking `nil` is that you won't call `current_start_mode` unless is strictly needed because, if `changes[:start_mode]` is `nil`, there is nothing to compare.

Anyway, I've reimplemented the changes calculation mechanism so this line looks simpler now.